### ISSUE TITLE
OpenMPT's IT instruments >=0x7B are not working and other minor issues.

### ISSUE THREAD

**Issue by Honguito98** [libxmp/libxmp#104]
**Note:** tracks composed using all compatible mix levels from OpenMPT
Test case includes .it files as well affected soundtracks in ogg/vorbis/flac codec.

```
Password to decompress zip file: NDUTQ3ejR4NTJD1MyN3Y3g
Test case: https://drive.google.com/open?id=0B6WljCvFQYQkODdCMEE5MFB4SmM
```

```
===============================================
File: batcir.it
Created with: OpenMPT 1.26.05.00
> Observations:
* Minor issues with off-tune pitch bends


================================================
File: viewpoin.it
Created with: OpenMPT 1.26.05.00


Affected sub track: Menu song (and others?)
> Observations:
* Minor timing issues (maybe ramp-down/up routines, because bass drums sounds are smoothed, must be sharper)

Affected sub track: Ending & Credits (pattern position 0xA0)
> Observations:
* 'SAx' (weird?) command behavior - SAx should act when Oxx is set
* Auto vibrato does not work
* Bass instrument 0x7B (#123) not working

Affected sub track(s): New record, continue, and so on (just after ending song)
 also affected on track 0x0
> Observations:
* Instruments >= 0x7B are not working?
```

**Comment by AliceLR** [comment]
The test case URL is a dead link; can you reupload these?

**Comment by sezero** [comment]
Should we really work on an issue which relies on a password-protected zip (even if it gets re-uploaded)?

**Comment by AliceLR** [comment]
> Should we really work on an issue which relies on a password-protected zip (even if it gets re-uploaded)?

Yeah, I don't really get why it's password-protected. I'll see if I already have this downloaded somewhere...

Some of these (autovibrato) may have already been addressed, but it'd be nice to verify if there are real bugs here.

**Comment by Honguito98** [comment]
Hi.
It's been a long time since I posted this issue.

I do not remember exactly why I password-protected the file.

I think, probably because of copyright concerns in Google Drive, because they are rebuilds/arrangements of existing, old game songs, at least the IT modules and its patterns are of my creation, and its samples where recorded manually (i.e. they are not direct extract from original source), but I think still being a gray-area topic

I don't have anymore the exact zip file, but I could provide the IT modules, although they may behave slightly differently due to changes to the composition that I could made back then.


Maybe the best way to share them, is to build an isolated test case with the necessary sound samples and composition, to avoid potential copyright issues. Creating that files could take me a couple of days since I have little free time.



Let me know if you prefer the stripped-down version (sample and pattern count preserved, just set to zero-length / cleared), or the full version (with all samples and patterns), both password-less.

Kind regards
Honguito98

**Comment by Honguito98** [comment]
Hi,
Ok here are the IT files:
https://drive.google.com/drive/folders/13b4lylP52fgcEPX0ATg9rCUFNOVghdg6?usp=drive_link

I added another one that I had laying around, I did not test with XMP through, but I think may be useful.
From that module, just ignore the song 'Multiball' (pattern 53) is not finished and has missing sounds.


-Honguito98

**Comment by AliceLR** [comment]
> File: batcir.it
> Created with: OpenMPT 1.26.05.00
> > Observations:
> * Minor issues with off-tune pitch bends

This seems subjective—nothing sounded "wrong" to me. It's possible the issue was already fixed.
 
> * Minor timing issues (maybe ramp-down/up routines, because bass drums sounds are smoothed, must be sharper)

libxmp uses a different ramping/anticlick routine than OpenMPT; split to #969.

> * 'SAx' (weird?) command behavior - SAx should act when Oxx is set

This was fixed in 2015 AFAICT: https://github.com/libxmp/libxmp/commit/93d8dd8

The original report might have been based on an old version of libxmp at the time, or there might have been another bug preventing this effect from working, but these notes work now.
 
> * Auto vibrato does not work

This was most likely fixed (or at least worked around) in https://github.com/libxmp/libxmp/pull/218. It doesn't help for testing that this instrument has very light autovibrato.

> * Bass instrument 0x7B (#123) not working
> * Instruments >= 0x7B are not working?

The samples load fine but xmp-cli reports that the instruments don't have these samples linked. It seems the IT loader is wiping key map entries where the sample number is >=120: https://github.com/libxmp/libxmp/commit/24540887cd04fed745d0b27cde9599baec6262ca. I don't understand the relevance of this number, because Impulse Tracker has a 99 sample/instrument limit and MPT/OpenMPT support far more. I will just increase this bound to libxmp's actual limit (255).

(As a side note, no one wants to hunt for minor inaccuracies in a 209 position-long module—smaller modules that obviously reproduce the issue without other distractions are preferable and make triage much easier.)
